gsy_framework/sim_results/electric_blue/aggregate_results.py

```diff
@@ -130,6 +130,20 @@ def add_offer(self, offer: Dict) -> None:
         assert offer["seller_id"] == self.device_uuid, "Device is not seller of the offer."
         self.open_offers.append(offer)
 
+    @property
+    def average_sell_price(self) -> float:
+        """Return average sell price for the timeslot."""
+        if self.total_energy_sold == 0:
+            return 0
+        return self.total_earned_eur / self.total_energy_sold
```

Comment:
I have two comments here:
1. this method returns an "energy rate" in EUR/kWh. We decided in the past to at least stick to this nomenclature in the backend
2. I am not sure if this really is the average sell rate here. I would expect the average rate to be the arithmetic mean of the rates at each time slot (calculation 1) not the quotient of summed earned money and summed sold energy (calculation 2)
 ```python
# calculation 1
print((1/2 + 1/3 + 1/4)/3)
#0.3611111111111111

calculation 2
print((1+1+1)/(2+3+4))
# 0.333333333333333
 ```
If we can live with this and only need the average as an approximation, then leave it as is, but if the user needs exact data, we would need to change this. But then the calculation is not that easy implemented I guess.

Reply:
Thank you for your comments.

2- Actually, I was uncertain which calculation is correct, so I just picked the one that seemed most appropriate to me. I also had a very small chat about this with @spyrostz. Now, I'm ready to switch to the other calculation if you think this one is incorrect. Please let me know.

1- Can you explain more? Should I change any variable name? Do you have any suggestions for me?

Thanks

Reply:
> - Can you explain more? Should I change any variable name? Do you have any suggestions for me?

What I mean is here to rename the method to: `average_sell_energy_rate` or `average_sell_rate` and also in the variable in the calling code. Thanks

Reply:
Oh, sure! Thanks

Reply:
Done.

Reply:
Done.

Reply:
Thanks! But how about the calculation of the mean. @spyrostz you know the requirements better. Which "mean" is actually wanted?

Reply:
The reason that this method is named as `average_sell_price` is in order to match the badly-named UI element (which mentions "Price" while the unit is Euro / MWh). Therefore I agree with Hannes that it should be renamed to reflect the term "rate" and not use the term "price".
For the other topic, I am also not sure about the exact requirements here since I doubt that this chart is so thought through. I tend to agree with Hannes though, that probably what is required here is a timeseries of an average energy rate. In which case we need calculation #1 here.

Reply:
No worries then I can swtich to #1. But in order to do that, I need to include `sum_of_energy_rate`s and `number_of_trades` in the time series (and remove `price`). Do we all agree on that?

Reply:
I agree that it would be good to include both sum_of_energy_trades and number_of_trades, they are required in order to calculate this especially in longer duration / resolution options.